pkg/meta/model/table.go

```diff
@@ -1065,7 +1065,7 @@ func (pi *PartitionInfo) IDsInDDLToIgnore() []int64 {
 			return ids
 		}
 	case ActionDropTablePartition:
-		if len(pi.DroppingDefinitions) > 0 && pi.DDLState == StateDeleteOnly {
+		if len(pi.DroppingDefinitions) > 0 && (pi.DDLState == StateDeleteOnly || pi.DDLState == StateWriteOnly) {
```

Comment:
Do you think it's more clear to change the condition to `pi.DDLState != StatePublic`, or even remove this state filter? IMO Any partition state that started dropping should be ignored.

Reply:
For `pi.DDLAction == ActionDropTablePartition` the only `pi.DDLState`'s that are used are `StateWriteOnly`, `StateDeleteOnly` and `StateDeleteReorganization`, and for `StateDeleteReorganization`, `pi.DroppingDefinitions` is nil, so you are right, we can remove the check for `pi.DDLState`. Thanks!